airbyte-integrations/base-python/base.py

```diff
@@ -66,25 +76,30 @@ def start(self):
                 sys.exit(0)
 
             rendered_config_path = os.path.join(temp_dir, 'config.json')
-            config_object = source.read_config(parsed_args.config)
-            source.render_config(config_object, rendered_config_path)
+            raw_config = source.read_config(parsed_args.config)
+            rendered_config = source.transform_config(raw_config)
+            source.write_config(rendered_config, rendered_config_path)
+
+            config_container = ConfigContainer(
+                raw_config=raw_config,
+                rendered_config=rendered_config,
+                raw_config_path=parsed_args.config,
+                rendered_config_path=rendered_config_path)
 
-            # todo: output message for check
             if cmd == "check":
-                check_result = source.check(logging, rendered_config_path)
+                check_result = source.check(log_line, config_container)
                 if check_result.successful:
-                    print("Check succeeded")
+                    log_line("Check succeeded", "INFO")
                     sys.exit(0)
                 else:
-                    print("Check failed")
+                    log_line("Check failed", "ERROR")
                     sys.exit(1)
             elif cmd == "discover":
-                schema = source.discover(logging, rendered_config_path)
-                print(schema.schema)
+                catalog = source.discover(log_line, config_container)
+                print(catalog.serialize())
                 sys.exit(0)
             elif cmd == "read":
-                # todo: pass in state
-                generator = source.read(logging, rendered_config_path)
+                generator = source.read(log_line, config_container, parsed_args.catalog, parsed_args.state)
```

Comment:
Should we have have a similar "container" for catalog and state?

Reply:
will do this in a separate PR